model/SearchEngine/Driver/Elasticsearch/ElasticSearch.php

```diff
@@ -0,0 +1,238 @@
+                    throw new SyntaxException($queryString, $message);
+                default:
+                    $message = 'An unknown error occurred during search';
+                    $this->logger->error(sprintf('Elasticsearch: %s "%s"', $message, $exception->getMessage()));
+                    throw new SyntaxException($queryString, __($message));
+            }
+        }
+    }
+
+    public function index($documents = []): int
+    {
+        $documents = $documents instanceof Iterator
+            ? $documents
+            : new ArrayIterator($documents);
+
+        return $this->indexer->buildIndex($documents);
+    }
+
+    public function remove($resourceId): bool
+    {
+        return $this->indexer->deleteDocument((string)$resourceId);
+    }
+
+    public function supportCustomIndex(): bool
+    {
+        return true;
+    }
+
+    public function createIndexes(): void
+    {
+        $indexFile = $this->getIndexFile();
+
+        $indexes = [];
+
+        if ($indexFile && is_readable($indexFile)) {
+            $indexes = require $indexFile;
+        }
+
+        foreach ($indexes as $index) {
+            $index['index'] = $this->prefixer->prefix($index['index']);
+
+            $this->client->indices()->create($index);
+        }
+    }
+
+    public function flush(): array
+    {
+        return $this->client->indices()->delete(
+            [
+                'index' => implode(
+                    ',',
+                    $this->prefixer->prefixAll(IndexerInterface::AVAILABLE_INDEXES)
+                ),
+                'client' => [
+                    'ignore' => 404
+                ]
+            ]
+        );
+    }
+
+    public function ping(): bool
+    {
+        return $this->client->ping();
+    }
+
+    private function buildResultSet(array $elasticResult = []): ResultSet
+    {
+        $uris = [];
+        $total = 0;
+
+        if ($elasticResult && isset($elasticResult['hits'])) {
+            foreach ($elasticResult['hits']['hits'] as $document) {
+                $document['_source']['id'] = $document['_id'];
+                $uris[] = $document['_source'];
+            }
+
+            $total = is_array($elasticResult['hits']['total'])
+                ? $elasticResult['hits']['total']['value']
+                : $elasticResult['hits']['total'];
+        }
+
+        return new ResultSet($uris, $total);
+    }
+
+    public function __toPhpCode()
+    {
+        return __CLASS__;
+    }
+
+    private function getIndexFile(): string
+    {
+        return $this->indexFile ?? __DIR__ .
+            DIRECTORY_SEPARATOR .
+            '..' .
+            DIRECTORY_SEPARATOR .
+            '..' .
+            DIRECTORY_SEPARATOR .
+            '..' .
+            DIRECTORY_SEPARATOR .
+            '..' .
+            DIRECTORY_SEPARATOR .
+            'config' .
+            DIRECTORY_SEPARATOR .
+            'index.conf.php';
```

Comment:
May be used implode to avoid repeating `DIRECTORY_SEPARATOR` and reduce the amount of code a bit (while keeping the PSR12 compliant)

```suggestion
        return $this->indexFile ?? implode(
            DIRECTORY_SEPARATOR,
            [__DIR__, '..', '..', '..', '..', 'config', 'index.conf.php']
        );
```